horace_core/sqw/file_io/@dnd_binfile_common/put_image.m

```diff
@@ -0,0 +1,36 @@
+function obj = put_image(obj, s, e)
+%PUT_IMAGE_SIGNAL Write the image signal and error to file
+%
+% Input:
+% ------
+% s     The image signal. Must be a numeric array.
+% e     The image error. This must be numeric array and have the same number of
+%       elements as s.
+%
+BYTES_IN_SINGLE = 4;
+
+check_obj_initiated_properly_(obj);
+
+validateattributes(s, {'numeric'}, {}, 1);
+validateattributes(e, {'numeric'}, {}, 2);
+
+if numel(s) ~= numel(e)
+    error( ...
+        'HORACE:DND_BINFILE_COMMON:length_mismatch', ...
+        'Input image signal and error have different numbers of elements.' ...
+    );
+end
+
+bytes_to_write = numel(s)*BYTES_IN_SINGLE;
+bytes_in_file = obj.e_pos_ - obj.s_pos_;
+
+if bytes_to_write ~= bytes_in_file
+    error( ...
+        'HORACE:DND_BINFILE_COMMON:invalid_length', ...
+        ['Image signal size in file does not match the size of the input.\n' ...
+         '''%i'' elements in file, ''%i'' elements in input array.'], ...
+        bytes_in_file/BYTES_IN_SINGLE, numel(bytes_to_write) ...
+    );
+end
```

Comment:
It can be overwriting an existing file, or writing a new one. When you initialize a loader you initialize with an object and a file path (existing or new). I think `obj.s_pos_` and `obj.e_pos_` are derived from the object, and hence so is the allowed image array size.

I've tried to clear this up in the function help